db-backup-files/2-patternatlas-data-inserts.sql

```diff
@@ -263,29 +264,29 @@ INSERT INTO public.pattern VALUES ('678a2e5a-697f-4416-afe6-fd18bc935bdc', 'Hybr
es/af7780d5-1f97-4536-8da7-4194b093ab1d/bcd4c7a1-3c92-4f8c-a530-72b8b95d3750)): \nSince a set of data values is represented in superposition, the data values can be manipulated at once (using quantum parallelism). \nFurthermore, multiple arithmetic operations (e.g., addition or multiplication) for [Basis Encoding](pattern-languages/af7780d5-1f97-4536-8da7-4194b093ab1d/bcd4c7a1-3c92-4f8c-a530-72b8b95d3750) are known that can also be applied to values in superposition.\n \n\nNote that algorithms that specify the usage of a QRAM share the following assumption: \nState preparation via the QRAM is efficient and, therefore, of logarithmic runtime [(Schuld and Petruccione 2018)](https://www.springer.com/gp/book/9783319964232). \nHowever, to our best knowledge, there are currently no commercial hardware implementations for QRAM.\nThus, a different state preparation routine has to load the data in the specified encoding. \nA major disadvantage today is that currently no state preparation routine for an arbitrary state is known to be as efficient as the logarithmic runtime of a QRAM. \nAs a result, a theoretically exponential speed-up of an algorithm using QRAM is only possible if the state preparation can be realized by an efficient state preparation method. \n\n", "Context": "For accessing the values of input data, a random access memory is needed.\n\n", "Solution": "When a classical random access memory (RAM) gets an address to a memory index, it transfers the data value stored at this address into a specified output register. \nThe functionality of quantum random access memory (QRAM) is similar, however, the registers are not classical but quantum registers [(Johnston, Harrigan and Gimeno-Segovia 2019)](https://www.oreilly.com/library/view/programming-quantum-computers/9781492039679/). \nConsequently, the address and output registers can be in  superposition instead of only single values.\nThe solution sketch demonstrates the loading process of a QRAM [(Giovannetti, Lloyd and Maccone 2008)](https://arxiv.org/abs/0708.1879) holding a superposition of the first two addresses ($\\newcommand{\\state}[1]{{\\left| #1 \\right>}} \\frac{1}{\\sqrt{2}}\\state{00} + \\frac{1}{\\sqrt{2}}\\state{01})$ as input register. \nLoading the data values of the corresponding addresses results in the following state:\n\n\n$$\n\\newcommand{\\colVec}[1]{% inline column vector\n\\bigl( \\begin{smallmatrix}#1\\end{smallmatrix}\\bigr)\n}\n\\newcommand{\\bigColVec}[1]{% inline column vector\n\\left( \\begin{matrix}#1\\end{matrix}\\right)\n}\n\\newcommand{\\state}[1]{{\\left| #1 \\right>}}\\state{\\psi} = \\frac{1}{\\sqrt{2}} \\state{00}\\state{010} + \\frac{1}{\\sqrt{2}}\\state{01}\\state{110}\n$$  \n\n\nLoading $m$ of $n$ data values using a QRAM can be generalized as follows [(Schuld and Petruccione 2018)](https://www.springer.com/gp/book/9783319964232): \n\n\n$$\n\\newcommand{\\colVec}[1]{% inline column vector\n\\bigl( \\begin{smallmatrix}#1\\end{smallmatrix}\\bigr)\n}\n\\newc
```

Comment:
in the paper it is stated that QPE uses Matrix Encoding and on the other hand that Matrix Encoding can be used to define a unitary operation as input for QPE. 

Here, currently, Matrix Encoding canBeUsedWith QPE is defined in the content. @lharzenetter what do you think?